nmaipy/parcels.py

```diff
@@ -216,6 +225,30 @@ def filter_features_in_parcels(features_gdf: gpd.GeoDataFrame, config: Optional[
         no_parent = (gdf.parent_id == "") | gdf.parent_id.isna()
         gdf = gdf.loc[~parent_removed | no_parent]
 
+    building_style_ids = DEFAULT_FILTERING["building_style_filtering"].keys()
+    out_gdf = []
+    for aoi_id in aoi_gdf[AOI_ID_COLUMN_NAME].unique():
+        gdf_aoi = gdf[gdf[AOI_ID_COLUMN_NAME] == aoi_id]
+        if len(gdf_aoi) == 0:
+            continue
+        gdf_aoi_buildings = gdf_aoi[gdf_aoi.class_id.isin(building_style_ids)]
+        if len(gdf_aoi_buildings) == 0:
+            out_gdf.append(gdf_aoi)
+            continue
+        aoi_poly = aoi_gdf[aoi_gdf[AOI_ID_COLUMN_NAME] == aoi_id].to_crs(AREA_CRS[region]).iloc[0].geometry
+        building_statuses = []
+        for building_poly in gdf_aoi_buildings.to_crs(AREA_CRS[region]).geometry:
+            building_status = nmaipy.reference_code.get_building_status(building_poly, aoi_poly)
+            building_statuses.append(building_status)
+        building_statuses = pd.DataFrame(building_statuses)
+        building_statuses.index = gdf_aoi_buildings.index
+        gdf_aoi_buildings = pd.concat([gdf_aoi_buildings, building_statuses], axis=1)
+        gdf_aoi_buildings = gdf_aoi_buildings[gdf_aoi_buildings.building_keep].drop(columns=["building_keep"])
+        gdf_aoi = pd.concat([gdf_aoi[~gdf_aoi.class_id.isin(building_style_ids)], gdf_aoi_buildings], ignore_index=True)
+        out_gdf.append(gdf_aoi)
+
+    if len(out_gdf) > 0:
+        gdf = pd.concat(out_gdf)
```

Comment:
There is a bug here where you return the unfiltered gdf accidentally.

```suggestion
    if len(out_gdf) > 0:
        gdf = pd.concat(out_gdf)
    else:
        gdf = gpd.GeoDataFrame(columns=gdf.columns)
```

Reply:
Thanks, good catch